packages/web3-eth-ens/src/resolver.ts

```diff
@@ -0,0 +1,125 @@
+import { Address, sha3, isHexStrict } from 'web3-utils';
+import { Contract, NonPayableCallOptions } from 'web3-eth-contract';
+import { inputAddressFormatter } from 'web3-common';
+import { interfaceIds, methodsInInterface } from './config';
+import { Registry } from './registry';
+import { RESOLVER } from './abi/resolver';
+import { namehash } from './utils';
+
+//  Default public resolver
+//  https://github.com/ensdomains/resolvers/blob/master/contracts/PublicResolver.sol
+
+export class Resolver {
+	private readonly registry: Registry;
+
+	public constructor(registry: Registry) {
+		this.registry = registry;
+	}
+
+	private async getResolverContractAdapter(ENSName: string) {
+		//  TODO : (Future 4.1.0 TDB) cache resolver contract if frequently queried same ENS name, refresh cache based on TTL and usage, also limit cache size, optional cache with a flag
+		return this.registry.getResolver(ENSName);
+	}
+
+	//  https://eips.ethereum.org/EIPS/eip-165
+	// eslint-disable-next-line class-methods-use-this
+	public async checkInterfaceSupport(
+		resolverContract: Contract<typeof RESOLVER>,
+		methodName: string,
+	) {
+		if (interfaceIds[methodName] === undefined) throw new Error(); //  ResolverMethodMissingError(resolver address, methodName);
+
+		let supported = false;
+		try {
+			supported = (await resolverContract.methods
+				.supportsInterface(interfaceIds[methodName])
+				.call()) as Awaited<Promise<boolean>>;
```

Comment:
If contract typing is correct, then we don't need to use `as` here. Please check what type error you were getting here.

Reply:
`error  Unsafe assignment of an any value  @typescript-eslint/no-unsafe-assignment`

Reply:
@jdevcs  That's not right. Please create an issue to address this later.

Reply:
I was also wondering with typed contract feature it should have correct type.